flang/lib/Semantics/resolve-directives.cpp

```diff
@@ -2173,7 +2178,7 @@ void OmpAttributeVisitor::CreateImplicitSymbols(
       // TODO 5) dummy arg in orphaned taskgen construct -> firstprivate
       if (prevDSA == Symbol::Flag::OmpShared) {
         // 6) shared in enclosing context -> shared
-        makeSharedSymbol();
+        makeSharedSymbol(/*setFlag=*/false);
```

Comment:
With these set I encountered various test failures for variables used in clauses which cannot be combined with `shared`

Reply:
OK, I guess adding `OmpShared` in all cases would require considerable extra work.
It would also break this patch's assumption, that `OmpShared` means a `shared` clause was used.

So why don't we set `OmpShared` only in the `if (dsa.has_value())` case?
This should handle all situations properly:
- `!$omp parallel shared(i)` - don't privatise
- `!$omp parallel` | `!$omp parallel default(shared)` - privatise (assuming the standard disallows `default` to override predetermined loop induction variables DSA)

Then a symbol with `OmpShared` would always mean it was explicitly made shared.